### DISCUSSION TITLE
sep.extract parameters of unconvolved data when using filter_kernel

### DISCUSSION THREAD

**Discussion by cosmic-nori** [sep-developers/sep#132] [Q&A] [answered]
I am wondering how does the `sep.extract()` obtain object parameters of unconvolved data
(e.g., `peak` and `tnpix` parameters) when using filter_kernel. Could you please tell me about that?

I have not seen that a detected object with `tnpix` parameter = 0 so far.
But I guess it is possible that such an object can be detected. 
Or, if we set `minarea` = 1 when using filter_kernel, is the criteria also applied to the unconvolved data? Thank you.

**Comment by kbarbary** [comment, marked as answer]
I believe how it works is: for each object, `sep.extract` stores member pixel information. For each pixel, both the unconvolved and convolved values are stored. Once all the pixels belonging to the object have been recorded, `peak` and `tnpix` are determined by going through the stored pixel values and doing various statistics on them.

`minarea` applies to the convolved data, I believe.

Yes, it should be possible to have `tnpix = 0`. You might be able to construct such a case by setting `minarea=1` and having an object with a few pixels just below the threshold (such that one convolved pixel is just above the threshold).

**Reply by cosmic-nori** [reply]
Thank you for the clear explanation about `sep.extract`.
I understand that. After that, I tried to construct the case by setting `minarea=1`
and low `thresh` value of 0.5. Finally I could find objects with `tnpix=0` as you suggested.
I was able to figure things out, thanks again!
